tools/codegen/api/cpp.py

```diff
@@ -201,9 +202,12 @@ def returntype_type(t: Type, *, mutable: bool) -> CType:
         elif t.name == BaseTy.Scalar:
             return BaseCType(scalarT)
     elif isinstance(t, ListType):
-        elem = returntype_type(t.elem, mutable=mutable)
+        elem = returntype_type(t.elem, mutable=False)
```

Comment:
maybe you want to assert that mutable is false here?

Reply:
yep, thanks! (this was a relic of me originally trying to return `Tensor(a!)[]` in the first version of the PR, but now these functions don't return anything).